lib/proxy.js

```diff
@@ -112,6 +113,17 @@ var server = http.createServer(function(req, res) {
       buildConfig: result.buildConfig,
     };
 
+    if (result.proxy.basicAuth) {
+      let creds = result.proxy.basicAuth;
+      if (creds.username && creds.password) {
+        let user = auth(req);
+        if (!user || (user.name !== creds.username || user.pass !== creds.password)) {
```

Comment:
This does not work if you use numbers for your username or password. We should convert everything to a string before comparing. You can get this to fail by setting your password to `123` in the yaml file.